### ISSUE TITLE
Expose io:Error from from_filepath

### ISSUE THREAD

**Issue by alexanderkjall** [mbrubeck/tree_magic#7]
The from_filepath currently just swallows any errors that happens in regards to the file that was sent in as a parameter. This makes it hard to give the end user a good error message if something goes wrong.

Maybe consider to change this function from an Option to a Result?

This is how I used it now, and had to write a very generic error message: https://github.com/YaLTeR/wl-clipboard-rs/pull/16/files#diff-222c1b6ee536abb8fc978efb87e55b43402290cacf20cfa4a296e4373d1461c1R464

**Comment by mbrubeck** [comment]
I agree, `from_filepath` and related functions  should return `Result`.

Patches are welcome, if someone else gets to this before I do.
